backend/src/agents/web_agent.py

```diff
@@ -38,8 +38,8 @@ async def web_general_search_core(search_query, llm, model) -> str:
                 }
             return json.dumps(response, indent=4)
         logger.info(f'Answer found successfully {answer_result}')
-        valid_answer = json.loads(answer_result["response"]).get("is_valid", "")
-        if valid_answer:
+        perform_web_search = json.loads(answer_result["response"]).get("perform_web_search", "")
```

Comment:
this should be `should_perform_web_search` too I think

Reply:
variable name not changed

Reply:
@evpearce - Done..